daemons/fenced/fenced_history.c

```diff
@@ -257,7 +259,23 @@ stonith_xml_history_to_list(xmlNode *history)
         op->completed_nsec = completed_nsec;
         crm_element_value_int(xml_op, F_STONITH_STATE, &state);
         op->state = (enum op_state) state;
-        stonith__xe_get_result(xml_op, &op->result);
+
+        /* @COMPAT We can't use stonith__xe_get_result() here because
+         * fencers <2.1.3 didn't include results, leading it to assume an error
+         * status. Instead, set an unknown status in that case.
+         */
+        if ((crm_element_value_int(xml_op, XML_LRM_ATTR_RC, &exit_status) < 0)
```

Comment:
This looks correct to me.

An alternative to avoid duplication here might be to add a boolean flag argument to stonith__xe_get_result(), telling it not to look for a legacy RC and to use indeterminate/unknown... if skipping the legacy RC fetch is the only special treatment needed for history ops in this case. May have to also bypass the null-check of the result argument based on the flag, not sure -- we must have made it past the null-check in RHBZ#2086230.

As you mentioned in the RHBZ, this would complicate get_result() a bit. It doesn't matter to me either way.

Reply:
Yeah I was on the fence between the two approaches, but doing it in stonith__xe_get_result() looked a bit hacky and would make every other caller less readable just to benefit this one.